src/io/sb3/fromSb3.ts

```diff
@@ -1,67 +1,71 @@
 import * as JSZip from "jszip";
-import * as sb3 from "../sb3";
+import * as sb3 from "./interfaces";
 
 import { OpCode } from "../../OpCode";
 
 import Block, { BlockBase } from "../../Block";
 import * as BlockInput from "../../BlockInput";
 import Costume from "../../Costume";
-import List from "../../List";
 import Project from "../../Project";
 import Sound from "../../Sound";
 import { Sprite, Stage } from "../../Target";
-import Variable from "../../Variable";
+import { List, Variable } from "../../Data";
 import Script from "../../Script";
 
-type getAsset = (info: {
+type getAssetData = (info: {
   type: "costume" | "sound";
   name: string;
   md5: string;
   ext: string;
   spriteName: string;
 }) => Promise<any>;
 
-function extractCostumes(target: sb3.Target, getAsset: getAsset): Promise<Costume[]> {
+function extractCostumes(target: sb3.Target, getAssetData: getAssetData): Promise<Costume[]> {
   return Promise.all(
     target.costumes.map(
       async (costumeData: sb3.Costume) =>
         new Costume({
           name: costumeData.name,
-          md5: costumeData.assetId,
-          ext: costumeData.dataFormat,
-          bitmapResolution: 2,
-          centerX: costumeData.rotationCenterX * 2,
-          centerY: costumeData.rotationCenterY * 2,
-          asset: await getAsset({
+          dataFormat: costumeData.dataFormat,
+          data: await getAssetData({
             type: "costume",
             name: costumeData.name,
             md5: costumeData.assetId,
             ext: costumeData.dataFormat,
             spriteName: target.name
-          })
+          }),
+
+          md5: costumeData.assetId,
+          ext: costumeData.dataFormat,
+
+          bitmapResolution: costumeData.bitmapResolution || 1,
+          centerX: costumeData.rotationCenterX || 0,
+          centerY: costumeData.rotationCenterY || 0
```

Comment:
I believe 3.0 automatically calculates rotation centers for costumes which don't have them, but that could get messy very quickly (e.g. calculating SVG rotation centers requires calculating the costume's bounds). Maybe add a note for now that this doesn't match 3.0's behavior.

Reply:
Or just allow `centerX` and `centerY` to be nullable and pass the complexity onto the Scratch implementation that uses `sb-edit`